internal/state/atomic.go

```
@@ -0,0 +1,55 @@
+	"os"
+	"path/filepath"
+)
+
+// writeAtomic writes data to a file atomically by using a temp file + rename.
+// relPath is relative to stateDir.
+func writeAtomic(stateDir, relPath string, data []byte, mode os.FileMode) error {
+	fullPath := filepath.Join(stateDir, relPath)
+	dir := filepath.Dir(fullPath)
+
+	// Ensure directory exists
+	os.MkdirAll(dir, 0700)
```

golangci-lint (2.12.2)

[error] Error return value of `os.MkdirAll` is not checked

(errcheck)